### PR TITLE
Delta support for Spark-3.4

### PR DESCRIPTION
**Pull request by cfmcgrady**
### _Why are the changes needed?_

test against delta-2.4.0

https://github.com/delta-io/delta/releases/tag/v2.4.0

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by bowenliang123 — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -2159,11 +2159,18 @@
                 <module>extensions/spark/kyuubi-spark-connector-kudu</module>
             </modules>
             <properties>
+                <delta.version>2.4.0rc1</delta.version>
                 <spark.version>3.4.0</spark.version>
                 <!-- FIXME: used for constructing Iceberg artifact name, correct it once Iceberg supports Spark 3.4 -->
                 <spark.binary.version>3.3</spark.binary.version>
-                <maven.plugin.scalatest.exclude.tags>org.scalatest.tags.Slow,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.HudiTest,org.apache.kyuubi.tags.IcebergTest</maven.plugin.scalatest.exclude.tags>
+                <maven.plugin.scalatest.exclude.tags>org.apache.kyuubi.tags.HudiTest,org.apache.kyuubi.tags.IcebergTest</maven.plugin.scalatest.exclude.tags>
```
Comment (pan3793) [inline pom.xml:2166]: there is no hudi test anymore

Comment (cfmcgrady) [reply pom.xml]: removed.

File: pom.xml [hunk 2]
```diff
@@ -2159,10 +2159,11 @@
                 <module>extensions/spark/kyuubi-spark-connector-kudu</module>
             </modules>
             <properties>
+                <delta.version>2.4.0</delta.version>
                 <spark.version>3.4.0</spark.version>
                 <!-- FIXME: used for constructing Iceberg artifact name, correct it once Iceberg supports Spark 3.4 -->
                 <spark.binary.version>3.3</spark.binary.version>
-                <maven.plugin.scalatest.exclude.tags>org.scalatest.tags.Slow,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.HudiTest,org.apache.kyuubi.tags.IcebergTest</maven.plugin.scalatest.exclude.tags>
+                <maven.plugin.scalatest.exclude.tags>org.apache.kyuubi.tags.IcebergTest</maven.plugin.scalatest.exclude.tags>
```
Comment (pan3793) [inline pom.xml:2166]:
```suggestion
                <maven.plugin.scalatest.exclude.tags>org.scalatest.tags.Slow,org.apache.kyuubi.tags.IcebergTest</maven.plugin.scalatest.exclude.tags>
```

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**